front-end/src/app/shared/utils/reatt-redes/redesignated.utils.ts

```diff
@@ -1,9 +1,23 @@
 import { ReattRedesTypes } from './reatt-redes.utils';
 import { SchBTransaction } from '../../models/schb-transaction.model';
+import { MemoText } from 'app/shared/models/memo-text.model';
 
 export class RedesignatedUtils {
   public static overlayTransactionProperties(transaction: SchBTransaction, activeReportId?: string): SchBTransaction {
     if (!transaction.reattribution_redesignation_tag) {
+      if (transaction.expenditure_purpose_descrip) {
+        const prefix = `[Original purpose description: ${transaction?.expenditure_purpose_descrip}] `;
+        if (transaction.memo_text) {
+          transaction.memo_text.text_prefix = prefix;
+          transaction.memo_text.text4000 = prefix + transaction?.memo_text?.text4000;
+        } else {
+          transaction.memo_text = MemoText.fromJSON({
+            text_prefix: prefix,
+            text4000: prefix,
+          });
+        }
+      }
```

Comment:
This block of code is very similar to the comparable chunk in rattributted.utils.ts. Could we consolidate the logic into a static function in reatt-redes.utils.ts? The new function would get passed the transaction and either the flavor of purpose_descrip it needed.

Reply:
Done!